### ISSUE TITLE
uploadAssetFromUrl hangs indefinitely

### ISSUE THREAD

**Issue by kentico-michaelb** [kontent-ai/management-sdk-js#82]
### Brief bug description

When running the sample .uploadAssetFromUrl method from the README, the script hangs and never gets a response from the API nor does it timeout.

### Repro steps/Test code

```
import { ManagementClient } from '@kontent-ai/management-sdk';
import {  createDeliveryClient } from '@kentico/kontent-delivery';

const client = new ManagementClient({
  projectId: '<project id here>',
  apiKey: '<MAPI key here>'
});

async function assetTest(){
  const imageAsset = await client
      .uploadAssetFromUrl()
      .withData({
          asset: {
              descriptions: [{ description: 'Image poster for warrior', language: { codename: 'en-US' } }],//codename for testing
              title: 'Warrior cover'
          },
          binaryFile: {
              filename: 'warrior.jpg'
          },
          fileUrl: 'https://upload.wikimedia.org/wikipedia/en/e/e3/Warrior_Poster.jpg'
      })
      .toPromise();
  
      return imageAsset
}

const log = await assetTest()

console.log(log)
```

### Expected behavior

Successful upload of asset, error response from API, or timeout.

### Test environment

 - @kontent-ai/management-sdk": "^3.0.0

### Additional context

Also impacts a customer project.

**Comment by Enngage** [comment]
I wasn't able to reproduce this issue. In fact, I copied your code and it's uploading file from the wikimedia correctly. See example - https://stackblitz.com/edit/typescript-fxj4cr?file=index.ts 

Can you reproduce the issue on the stackblitz?

**Comment by kentico-michaelb** [comment]
Getting error:

_Executing 'https://manage.kontent.ai/v2/projects/9997768f-b524-00d0-28d7-e0b72b97b51f/assets' failed. Request was retried '0' times.
Error: Request failed with status code 404_

when I input my Project ID and MAPI key on StackBlitz, and the same hanging behavior in VSCode when I copy and paste your version of the code, then run it using _npm run start_ which targets _node index.js_:

```
import { ManagementClient } from '@kontent-ai/management-sdk';
import {  createDeliveryClient } from '@kentico/kontent-delivery';

const client = new ManagementClient({
  projectId: '<project id here>', // id of your Kentico Kontent project
  apiKey: '<MAPI key here>'
});


async function assetTest() {
    const imageAsset = await client
      .uploadAssetFromUrl()
      .withData({
        asset: {
          descriptions: [
            {
              description: 'Image poster for warrior',
              language: { codename: 'en-US' },
            },
          ], //codename for testing
          title: 'Warrior cover',
        },
        binaryFile: {
          filename: 'warrior.jpg',
        },
        fileUrl:
          'https://upload.wikimedia.org/wikipedia/en/e/e3/Warrior_Poster.jpg',
      })
      .toPromise();
  
    return imageAsset;
  }
  
  const run = async () => {
    const log = await assetTest();
    console.log(log);
  };
  
  run();
```

**Comment by Enngage** [comment]
@kentico-michaelb thanks, but I honestly don't know how to reproduce it. I just tested it again and all is working correctly:

![success](https://user-images.githubusercontent.com/7137351/180441913-fc0b050f-96eb-4638-ac43-49fd449fe949.png)

**Comment by Samdez** [comment]
Hi @Enngage @kentico-michaelb I'm the programmer who reported the issue to Michael yesterday. A coworker of mine and I have been trying to work on it, but we can't get a response from API. Weirdly enough, the code on stackblitz works when ran from the browser, but when ran locally it does not. We've been trying with different Node versions (16.6.0 and 17.9.1). Are we missing something here? Thanks for your help

**Comment by Enngage** [comment]
Hey @Samdez , @kentico-michaelb ,

I was able to find the root cause (mutation of request config which caused additional headers to be added to request which then caused the process to stop..). I've released a fix in v`3.0.1`
